### PR TITLE
Port from Typed-Racket to Racket

### PR DESCRIPTION
**Pull request by akuhlens**
This pull request moves most of the compiler away from static type checking via typed racket to dynamic type checking via racket. This results in a 30-60x increase in build time performance. This pull request does not implement many of the dynamic enforcement of invariants that we will definitely want in the future.

### CODE REVIEW

**Comment by akuhlens** [comment]
Can we celebrate the fact that travis can now be relied upon for valuable feedback! 🎉

File: src/language/forms.rkt [hunk 1]
```diff
@@ -956,34 +1013,37 @@ class literal constants
      (CRec C)
      CVar))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 2]
```diff
@@ -956,34 +1013,37 @@ class literal constants
...
 
+#;
 (define-type Grift-Coercion
   (Rec C (U (Project Grift-Type Blame-Label)
             (Inject Grift-Type)
             (Sequence C C)
             (Mediating-Coercion C))))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 3]
```diff
@@ -956,34 +1013,37 @@ class literal constants
...
 
+#;
 (define-type Hyper-Coercion
   (HC Boolean Grift-Type (Option Blame-Label)
       Boolean Grift-Type
       (Mediating-Coercion Hyper-Coercion)))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 4]
```diff
@@ -956,34 +1013,37 @@ class literal constants
...
 
+#;
 (define-type Mixed-Coercion
   (U Grift-Coercion
      Hyper-Coercion
      (Mediating-Coercion Hyper-Coercion)))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 5]
```diff
@@ -956,34 +1013,37 @@ class literal constants
...
 
-(define-type Mixed-Coercion* (Listof Mixed-Coercion))
+#;(define-type Mixed-Coercion* (Listof Mixed-Coercion))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 6]
```diff
@@ -956,34 +1013,37 @@ class literal constants
...
 
-(define IDENTITY : Identity (Identity))
+(define IDENTITY (Identity))
 (define ID-EXPR (Quote-Coercion IDENTITY))
 (define ZERO-EXPR (Quote 0))
 
 
-(define-type Grift-Coercion* (Listof Grift-Coercion))
+#;(define-type Grift-Coercion* (Listof Grift-Coercion))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 7]
```diff
@@ -956,34 +1013,37 @@ class literal constants
...
 
-(define-type Data-Literal (U Integer Inexact-Real Char String))
+#;(define-type Data-Literal (U Integer Inexact-Real Char String))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 8]
```diff
@@ -956,34 +1013,37 @@ class literal constants
...
 
-(: data-literal? : Any -> Boolean : Data-Literal)
+#;(: data-literal? : Any -> Boolean : Data-Literal)
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 9]
```diff
@@ -997,16 +1057,18 @@ class literal constants
 
 ;; Represents the shallow tree structure of types where all subtrees
 ;; of the type are either and atomic type or a identifier for a type.
+#;
 (define-type Compact-Type
   (U (Fn Index (Listof Immediate-Type) Immediate-Type)
      (STuple Index (Listof Immediate-Type))
      (GRef Immediate-Type) (MRef Immediate-Type)
      (GVect Immediate-Type) (MVect Immediate-Type)))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 10]
```diff
@@ -997,16 +1057,18 @@ class literal constants
...
 
-(define-type Mu-Compact-Type (Mu Immediate-Type))
+#;(define-type Mu-Compact-Type (Mu Immediate-Type))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 11]
```diff
@@ -997,16 +1057,18 @@ class literal constants
...
 ;; Represent the shallow tree structure of coercions where all
 ;; subtrees of the type are either atomic types, the identity coercion
 ;; or coercion identifiers.
+#;
 (define-type Compact-Coercion
   (U (Project Immediate-Type Blame-Label)
      (Inject Immediate-Type)
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 12]
```diff
@@ -1021,9 +1083,9 @@ class literal constants
      (CTuple Index (Listof Immediate-Coercion))
      (Ref Immediate-Coercion Immediate-Coercion Proxied-Symbol)))
 
-(define-type Mu-Compact-Coercion (Mu Immediate-Coercion))
+#;(define-type Mu-Compact-Coercion (Mu Immediate-Coercion))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 13]
```diff
@@ -1021,9 +1083,9 @@ class literal constants
...
 
-(define-type Immediate-Type (U Atomic-Type (Static-Id Uid)))
+#;(define-type Immediate-Type (U Atomic-Type (Static-Id Uid)))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 14]
```diff
@@ -1036,8 +1098,9 @@ class literal constants
 ;; N^2 implementation of guarded reference coercions because
 ;; that may require injects to carry blame labels thus reducing
 ;; the impact of this optimization.
-(define-type Immediate-Coercion (U Identity (Static-Id Uid)))
+#;(define-type Immediate-Coercion (U Identity (Static-Id Uid)))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 15]
```diff
@@ -1036,8 +1098,9 @@ class literal constants
...
  
+#;
 (define-type Coercion/Immediate-Type
   (Rec C (U Identity
             (Failed Blame-Label)
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 16]
```diff
@@ -1050,28 +1113,29 @@ class literal constants
             (MonoVect Immediate-Type)
             (CTuple Index (Listof C)))))
 
-(define-type Coercion/Immediate-Type* (Listof Coercion/Immediate-Type))
+#;(define-type Coercion/Immediate-Type* (Listof Coercion/Immediate-Type))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 17]
```diff
@@ -1050,28 +1113,29 @@ class literal constants
...
 
+#;
 (define-type (Map-Bnd E1 E2)
   (case->
    [(Bnd Uid Grift-Type E1) -> (Bnd Uid Grift-Type E2)]
    [(Pair Uid E1) -> (Pair Uid E2)]))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 18]
```diff
@@ -1050,28 +1113,29 @@ class literal constants
...
 
-(: map-bnd (All (A B) (A -> B) -> (Map-Bnd A B)))
+#;(: map-bnd (All (A B) (A -> B) -> (Map-Bnd A B)))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 19]
```diff
@@ -1050,28 +1113,29 @@ class literal constants
...
 (define ((map-bnd f) b)
   (match b
     [(Bnd i t e) (Bnd i t (f e))]
     [(cons i e) (cons i (f e))]))
 
-(define-type Id (U Uid String))
+#;(define-type Id (U Uid String))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 20]
```diff
@@ -1050,28 +1113,29 @@ class literal constants
...
 
-(define-type VectorAccessMode (U 'check-bounds 'no-check-bounds))
+#;(define-type VectorAccessMode (U 'check-bounds 'no-check-bounds))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: src/language/forms.rkt [hunk 21]
```diff
@@ -1050,28 +1113,29 @@ class literal constants
...
 
-(define-type (Fun E) (Code Uid* E))
-(define-type (CLambda E) (Lambda Uid* (Castable (Option Uid) E)))
-(define-type (Bnd* E) (Listof (Pairof Uid E)))
+;; (define-type (Fun E) (Code Uid* E))
+;; (define-type (CLambda E) (Lambda Uid* (Castable (Option Uid) E)))
+;; (define-type (Bnd* E) (Listof (Pairof Uid E)))
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: commented code

File: tests/test-compile.rkt [hunk 22]
```diff
@@ -1,20 +1,23 @@
-#lang typed/racket
+#lang typed/racket/no-check
 ;; This is a redo of the basic compiler with interpreters between every micro compiler.
 ;; Once the file is finished compiling the program is run and the value checked against
 ;; an expected value. The real compiler is then invoked. And the result once again checked.
 
-(require "./rackunit.rkt"
+(require rackunit #;"./rackunit.rkt"
```
Comment (deyaaeldeen) [inline tests/test-compile.rkt]: commented code

File: tests/test-compile.rkt [hunk 23]
```diff
@@ -1,20 +1,23 @@
...
-(require/typed racket/control
-  (abort (Any -> Nothing)))
 
-(require/typed racket/exn
-  (exn->string (exn -> String)))
+;; (require/typed racket/control
+;;   (abort (Any -> Nothing)))
+
+;; (require/typed racket/exn
+;;   (exn->string (exn -> String)))
```
Comment (deyaaeldeen) [inline tests/test-compile.rkt]: commented code

File: src/language/primitives.rkt [hunk 24]
```diff
@@ -1,14 +1,136 @@
...
+  (Print              #t (String -> Unit))
+  (Exit               #t (Int -> Bottom)))
 
-(define-type Grift-Primitive (U Grift-Prim Grift-Prim!))
-(define-predicate grift-primitive? Grift-Primitive)
-
+#;
 (define-type Grift-Prim
   (U Assoc-Stack-Primitive
      Int->Int-Primitive
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: commented code

File: src/language/primitives.rkt [hunk 25]
```diff
@@ -30,325 +152,342 @@
...
      ->Char-Primitive))
 
-(define-predicate grift-prim? Grift-Prim)
-
-(define-type Assoc-Stack-Primitive
-  (U 'make-assoc-stack
-     'assoc-stack-find
-     'assoc-stack-ref
-     'assoc-stack-pop!))
+#;(define-predicate grift-prim? Grift-Prim)
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: commented code

File: src/language/primitives.rkt [hunk 26]
```diff
@@ -30,325 +152,342 @@
...
 
-(define-type Assoc-Stack!-Primitive
-  (U 'assoc-stack-set!
-     'assoc-stack-push!))
+;; (define-type Grift-Prim!
+;;   (U Assoc-Stack!-Primitive
+;;      Timer-Primitive
+;;      Char->Unit-Primitive
+;;      Int->Unit-Primitive
+;;      FloatxInt->Unit-Primitive))
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: commented code

File: src/language/primitives.rkt [hunk 27]
```diff
@@ -30,325 +152,342 @@
...
-     Char->Unit-Primitive
-     Int->Unit-Primitive
-     FloatxInt->Unit-Primitive))
-
-(define-predicate grift-prim!? Grift-Prim!)
+;; (define-predicate grift-prim!? Grift-Prim!)
 #;(: grift-prim!? (Any -> Boolean : Grift-Prim!))
 #;
 (define (grift-prim!? x)
   (or (timer-primitive? x)))
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: commented code

File: src/language/primitives.rkt [hunk 28]
```diff
@@ -30,325 +152,342 @@
...
-(define-type Int->Int-Primitive (U 'binary-not))
-(define-predicate Int->Int-primitive? Int->Int-Primitive)
+#;(define-type Int->Int-Primitive (U 'binary-not))
+#;(define-predicate Int->Int-primitive? Int->Int-Primitive)
 
+#;
 (define-type IntxInt->Int-Primitive (U '* '+ '-
                                        'binary-and 'binary-or 'binary-xor 
                                        '%/ '%>> '%<< '%%
                                        'quotient))
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: commented code

File: src/language/primitives.rkt [hunk 29]
```diff
@@ -30,325 +152,342 @@
...
-  (U 'fl+ 'fl- 'fl* 'fl/ 'flmodulo 'flexpt 'flmin 'flmax))
-(define-type FloatxFloat->Int-Primitive (U 'flquotient))
-(define-type Float->Float-Primitive
-  (U 'flabs 'flround 'flfloor 'flceiling 'fltruncate
-     'flsin 'flcos 'fltan 'flasin 'flacos 'flatan
-     'fllog 'flexp 'flsqrt
-     'flmin 'flmax 'flnegate))
-(define-type FloatxInt->Unit-Primitive (U 'print-float))
-;(define-type Float->Unit-Primitive (U 'print-float))
+#;(define-type IxI->I-Prim IntxInt->Int-Primitive)
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: commented code

File: src/language/primitives.rkt [hunk 30]
```diff
@@ -30,325 +152,342 @@
...
-     (read-float . ,->FLOAT-TYPE)
-     (print-float . ,FLOATxINT->UNIT-TYPE) 
-     (timer-start . ,->UNIT-TYPE)
-     (timer-stop . ,->UNIT-TYPE)
-     (timer-report . ,->UNIT-TYPE))))
+#;(define-predicate IntxInt->Int-primitive? IntxInt->Int-Primitive)
 
+;; (define-type ->Int-Primitive (U 'read-int))
+;; (define-type Int->Unit-Primitive (U 'print-int))
+;; (define-type ->I-Prim ->Int-Primitive)
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: commented code

File: src/language/primitives.rkt [hunk 31]
```diff
@@ -30,325 +152,342 @@
...
+;;      (flexp  . ,FLOAT->FLOAT-TYPE)
+;;      (flsqrt . ,FLOAT->FLOAT-TYPE)
+;;      (flexpt . ,FLOATxFLOAT->FLOAT-TYPE)
+;;      (float->int . ,FLOAT->INT-TYPE)
+;;      (int->float . ,INT->FLOAT-TYPE)
+;;      (read-float . ,->FLOAT-TYPE)
+;;      (print-float . ,FLOATxINT->UNIT-TYPE) 
+;;      (timer-start . ,->UNIT-TYPE)
+;;      (timer-stop . ,->UNIT-TYPE)
+;;      (timer-report . ,->UNIT-TYPE))))
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: commented code

File: src/language/primitives.rkt [hunk 32]
```diff
@@ -30,325 +152,342 @@
...
+
+(define (symbol->primitive s not-found-th)
+  (hash-ref symbol->primitive-table s not-found-th))
+
+#;
+(: grift-primitive->type
+   (-> Grift-Primitive (Fn Index (Listof Base-Type) Base-Type)))
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: commented code

File: src/language/primitives.rkt [hunk 33]
```diff
@@ -30,325 +152,342 @@
...
+     (define (err) (error 'grift-primitive "invalid: ~a" sym/prim))
+     (symbol->primitive sym/prim err)]
+    [else sym/prim]))
+
+(define (grift-primitive-effect? p)
+  (primitive-effectfull? (grift-primitive p)))
+
+(module+ test
+  (require rackunit)
+  (check-true (Fn? (grift-primitive->type 'flacos))))
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: I believe more tests needed.

File: src/language/primitives.rkt [hunk 34]
```diff
@@ -30,325 +152,342 @@
...
+;; (define-type Types-Prim (U 'Types-hashcons! 'Types-gen-index!))
+;; (define-type Print-Prim! (U 'Printf 'Print 'print-float
+;;                             'print-int 'print-char 'display-char))
+;; (define-type Bottom-Prim (U 'Exit))
+
+;; (define-type (UIL-Op E) (Op UIL-Prim (Listof E)))
+;; (define-type (UIL-Op! E) (Op UIL-Prim! (Listof E)))
+
+;; (define-type Grift-Primitive (U Grift-Prim Grift-Prim!))
+;; (define-predicate grift-primitive? Grift-Primitive)
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: commented code

File: src/casts/constants-and-codes.rkt [hunk 35]
```diff
@@ -1,13 +1,15 @@
...
 (require
  (for-syntax
+  racket/base
   racket/list
   racket/syntax
   syntax/parse
   "../language/forms.rkt"
   "../language/c-helpers.rkt")
- "../language/data0.rkt"
+ racket/list
```
[the remark this answers is not in the document]
Comment (akuhlens) [reply src/casts/constants-and-codes.rkt]: The `racket/list` on line 6 is at a different phase.

File: src/casts/constants-and-codes.rkt [hunk 36]
```diff
@@ -241,25 +243,21 @@
...
+           #`(define (func-alloc-name alloc-arg* ...)
                #,(cond
                    [(syntax->datum #'hashcons?)
                     #`(sr-hashcons-type
                        #,name-string tag-def `((field* . ,alloc-val*) ...))]
                    [else
                     #`(sr-alloc
                        #,name-string tag-def
-                      `(("index"    . ,(Quote 0))
+                       `(("index"    . ,(Quote 0))
```
[the remark this answers is not in the document]
Comment (akuhlens) [reply src/casts/constants-and-codes.rkt]: The extra space here fixes incorrect indention in the original.

File: src/casts/constants-and-codes.rkt (hunk 35 above)
[the remark this answers is not in the document]
Comment (deyaaeldeen) [reply src/casts/constants-and-codes.rkt]: right, did not see the closing paren.

File: src/casts/constants-and-codes.rkt (hunk 36 above)
[the remark this answers is not in the document]
Comment (deyaaeldeen) [reply src/casts/constants-and-codes.rkt]: right, i can not see for some reason :D

File: src/backend-c/code-generator.rkt [hunk 37]
```diff
@@ -103,7 +98,8 @@
             (unless (null? s)
               (display (car s))
               (display " ")
-              (loop (cdr s))))))))
+              (loop (cdr s)))
+            (display (default-flags)))))))
```
[the remark this answers is not in the document]
Comment (akuhlens) [reply src/backend-c/code-generator.rkt]: This `display` is writing to a string buffer. I have added a comment to this effect.

**Comment by akuhlens** [comment]
I propose that we keep type diffs in a graveyard file for the time being. The idea begin that I would like to eventually use them to derive contracts that can be applied to each pass for debugging purposes. I will put them and the current language definitions in a graveyard file for future reference.

File: src/casts/specify-representation.rkt [hunk 38]
```diff
@@ -798,26 +788,21 @@ but a static single assignment is implicitly maintained.
...
              [(data:DYN-CHAR-TAG) TYPE-CHAR-RT-VALUE]
              [else (op$ Print err-msg) (op$ Exit EXIT-FAILURE) UNDEF-IMDT]))]
-        [(Access (Dyn) 'immediate-value (app recur e) #f)
+        [(Dyn-Immediate-Value (app recur e))
          (op$ %>> e DYN-IMDT-SHIFT)]
-        [(Access (Dyn) 'immediate-tag (app recur e) #f)
-         (sr-get-tag e DYN-TAG-MASK)]
-        [(Access (Dyn) 'box-value (app recur e) #f)
+        [(Dyn-Immediate-Tag=Huh (app recur e) t)
+         (define (default-th) DYN-BOXED-TAG)
```
[the remark this answers is not in the document]
Comment (akuhlens) [reply src/casts/specify-representation.rkt]: The next clause uses the fact that it is a thunk to raise an error if that case is hit.

File: src/language/cast-or-coerce6.rkt [hunk 39]
```diff
@@ -19,63 +21,63 @@
...
-(struct (E) Closure-Caster form:simple-branch
-  ([arg : E])
+(struct Closure-Caster form:simple-branch
+  (arg)
   #:transparent)
 
-(struct (E) Closure-Ref form:leaf
-  ([arg : E] [key : Integer])
+(struct Closure-Ref form:leaf
+  (arg key)
```
[the remark this answers is not in the document]
Comment (akuhlens) [reply src/language/cast-or-coerce6.rkt]: I would like to write contracts for debugging purposes.

**Comment by akuhlens** [comment]
> One theme her is the commented code. I believe commented type signatures needs special handling because we need some sort of documentation about types that are no longer obvious. All other commented code blocks should be deleted.

I think ideally that would be the case, but we need documentation of what the types were and I am not willing to spend the time porting all the commented types to contracts right now. I think we should port comments to contracts as needed in the future.

I am fixing travis testing and then I hope to push what I have now. Please let me know if you have issues with this.

File: src/grift/type-check.rkt [hunk 40]
```diff
@@ -645,68 +648,75 @@ The type rules for core forms that have interesting type rules
...
   (match vect-ty
     [(Dyn) DYN-TYPE]
     [(MVect t) t]
     [otherwise (error 'type-check/mvector-val-type)]))
                       
 ;;; Procedures that destructure and restructure the ast
-(: tc-expr (-> G0-Ann-Expr Env (values G1-Ann-Expr Grift-Type)))
-(define (tc-expr exp env)
+;(: tc-expr (-> G0-Ann-Expr Env (values G1-Ann-Expr Grift-Type)))
+(define/contract (tc-expr exp env)
```
Comment (deyaaeldeen) [inline src/grift/type-check.rkt]: why keep the comment if you already created the contract?

File: src/language/forms.rkt [hunk 41]
```diff
@@ -670,47 +607,42 @@ class literal constants
...
-  (define mu-int : Grift-Type
+  (define mu-int
     (Mu (Scope (STuple 2 (list INT-TYPE (TVar 0))))))
   (test-equal? "Mu type" (consistent? mu-dyn mu-int) #t))
 
-(struct Bottom ([t1 : Grift-Type] [t2 : Grift-Type]) #:transparent)
-(define-type Grift-Type?! (U Bottom Grift-Type))
+(struct Bottom (t1 t2) #:transparent)
+
+;; Grift-Type?! (U Bottom Grift-Type)
```
Comment (deyaaeldeen) [inline src/language/forms.rkt]: I am not sure what is the purpose of this comment.

File: src/language/graveyard.rkt [hunk 42]
```diff
@@ -0,0 +1,1528 @@
+#lang typed/racket/no-check
+
```
Comment (deyaaeldeen) [inline src/language/graveyard.rkt]: I suggest to put the content here in a github issue and delete the file from the source tree.

File: src/language/grift.rkt [hunk 43]
```diff
@@ -0,0 +1,92 @@
...
+   (Grift-Lang/c
+    (Prog (list "" 0) (list (Ann2 (Observe ann-42 #f) no-src)))))
+  (define read-bool (Op 'read-bool '()))
+  (check-true (G0-Expr/c read-bool))
+  (check-true (grift-primitive? 'read-bool))
+
+  (define ann-f (Ann (Quote #f) (srcloc "lambda1.grift" 1 0 1 14)))
+  (check-true (G0-Ann-Expr/c ann-f))
+  
+  (define ann-lam1
```
Comment (deyaaeldeen) [inline src/language/grift.rkt]: why the 1 at the end?

File: src/language/make-begin.rkt [hunk 44]
```diff
@@ -18,6 +18,8 @@
        (Begin (append eff* value-eff*) value-value)]
       [(eff* value) (Begin eff* value)]))
 
+
+#;
```
Comment (deyaaeldeen) [inline src/language/make-begin.rkt]: why not just delete the expression?

File: src/language/primitives.rkt [hunk 45]
```diff
@@ -1,354 +1,156 @@
...
+(define-syntax (syntax->type stx)
+  (syntax-parse stx
+    [(_ (~or () (~datum Unit)))   #'UNIT]
+    [(_ (~datum Void)) #'VOID]
+    [(_ (~datum Int))  #'INT]
+    [(_ (~datum Float))  #'FLOAT]
+    [(_ (~datum String)) #'STRING]
+    [(_ (~datum Bool)) #'BOOL]
+    [(_ (~datum Char)) #'CHAR]
+    [(_ (~datum Obj))  #'OBJ]
```
Comment (deyaaeldeen) [inline src/language/primitives.rkt]: what is this type?

File: src/casts/convert-closures.rkt [hunk 46]
```diff
@@ -2231,5 +2180,61 @@ TODO We can generate better code in this pass for function casts.
...
+;; (assert-subtype? CoC4-Expr (U (U= CoC4-Expr) (U- CoC4-Expr)))
+
+;; ;; INVARIANT CoC4 = U= + U+
+;; ;;We check this by checking subtyping in both directions because it is
+;; ;;easier to know what is broken when this fails.
+;; (assert-subtype? (U (U= CoC5-Expr) (U+ CoC5-Expr)) CoC5-Expr)
+;; (assert-subtype? CoC5-Expr (U (U= CoC5-Expr) (U+ CoC5-Expr)))
+
+
+
```
[the remark this answers is not in the document]
Comment (deyaaeldeen) [reply src/casts/convert-closures.rkt]: I think the types defined in this module are important to keep, so I would suggest to still delete these comments and add the type definition to a github issue as I mentioned before.

File: src/language/cast-or-coerce6.rkt (hunk 39 above)
[the remark this answers is not in the document]
Comment (deyaaeldeen) [reply src/language/cast-or-coerce6.rkt]: I believe all commented code here should go to the github issue.

File: info.rkt [hunk 47]
```diff
@@ -12,3 +12,6 @@
     "src/backend-c/runtime/hashcons.o"
     "src/backend-c/runtime/runtime.o"))
 
+(define test-include-paths
```
[the remark this answers is not in the document]
Comment (akuhlens) [reply info.rkt]: It isn't yet; I will fix that.

File: src/language/make-begin.rkt (hunk 44 above)
Comment (akuhlens) [reply src/language/make-begin.rkt]: Done

File: src/language/grift.rkt (hunk 43 above)
Comment (akuhlens) [reply src/language/grift.rkt]: This is in a test sub-module. I am coming up with names for test cases.

File: src/casts/impose-cast-semantics.rkt [hunk 48]
```diff
@@ -1,23 +1,21 @@
-#lang typed/racket/base
+#lang typed/racket/base/no-check
+;;TODO rewrite the file in racket/base and remove type-racket imports
```
[the remark this answers is not in the document]
Comment (akuhlens) [reply src/casts/impose-cast-semantics.rkt]: This is still in type/racket/base/no-check

File: src/backend-c/code-generator.rkt [hunk 49]
```diff
@@ -1,14 +1,12 @@
...
-         "runtime-location.rkt")
+#lang typed/racket/no-check
+
+(require 
+ "../configuration.rkt"
+ "./generate-c.rkt"
+ syntax/location
+ (for-syntax racket/system)
+ (for-syntax "runtime-location.rkt")
+ "runtime-location.rkt")
```
[the remark this answers is not in the document]
Comment (akuhlens) [reply src/backend-c/code-generator.rkt]: This seems right to me.

**Pull request merged**